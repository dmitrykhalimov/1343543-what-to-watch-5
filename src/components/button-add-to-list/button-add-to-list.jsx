import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {addFavorite} from "../../store/api-actions";

const NotInList = {
  XLINK: `#add`,
  NAME: `My list`,
};

const InList = {
  XLINK: `#in-list`,
  NAME: `In list`
};

const ButtonAddToList = (props) => {
  const {id, isFavorite, isPromo, onFavoriteClick} = props;
  const handleClick = () => {
    onFavoriteClick(id, !isFavorite ? 1 : 0, isPromo);
  };

  return (
    <button
      className="btn btn--list movie-card__button"
      onClick={handleClick}
      type="button">
      <svg viewBox="0 0 19 20" width="19" height="20">
        <use xlinkHref={isFavorite ? InList.XLINK : NotInList.XLINK}></use>
      </svg>
      <span>{isFavorite ? InList.NAME : NotInList.NAME}</span>
    </button>
  );
};

ButtonAddToList.propTypes = {
  id: PropTypes.number.isRequired,
  isFavorite: PropTypes.bool.isRequired,
  isPromo: PropTypes.bool.isRequired,
  onFavoriteClick: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  onFavoriteClick(id, status, isPromo) {
    dispatch(addFavorite(id, status, isPromo));
  },
});

export {ButtonAddToList};
export default connect(null, mapDispatchToProps)(ButtonAddToList);
