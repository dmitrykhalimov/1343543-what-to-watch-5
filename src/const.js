export const ratingRanks = [
  {
    'Bad': 2.9
  },
  {
    'Normal': 4.9
  },
  {
    'Good': 7.9
  },
  {
    'Very good': 9.9
  },
  {
    'Awesome': 10
  }
];
