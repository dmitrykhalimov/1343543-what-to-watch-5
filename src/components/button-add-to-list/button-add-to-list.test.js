import React from "react";
import renderer from "react-test-renderer";
import {ButtonAddToList} from "./button-add-to-list";
import {MemoryRouter} from 'react-router-dom';
import {TEST_MOCKS} from "../../const";

// без MemoryRouter падает с ошибкой "You should not use <Link> outside a <Router>"

describe(`<ButtonAddToList /> render`, () => {
  it(`Should ButtonAddToList render correctly, isFavorite, isPromo`, () => {
    const tree = renderer
      .create(
          <MemoryRouter>
            <ButtonAddToList
              id={TEST_MOCKS.id}
              isFavorite={TEST_MOCKS.boolTrue}
              isPromo={TEST_MOCKS.boolTrue}
              onFavoriteClick={TEST_MOCKS.noop}
            />
          </MemoryRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`Should ButtonAddToList render correctly, !isFavorite, isPromo`, () => {
    const tree = renderer
      .create(
          <MemoryRouter>
            <ButtonAddToList
              id={TEST_MOCKS.id}
              isFavorite={TEST_MOCKS.boolFalse}
              isPromo={TEST_MOCKS.boolTrue}
              onFavoriteClick={TEST_MOCKS.noop}
            />
          </MemoryRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`Should ButtonAddToList render correctly, isFavorite, !isPromo`, () => {
    const tree = renderer
      .create(
          <MemoryRouter>
            <ButtonAddToList
              id={TEST_MOCKS.id}
              isFavorite={TEST_MOCKS.boolTrue}
              isPromo={TEST_MOCKS.boolFalse}
              onFavoriteClick={TEST_MOCKS.noop}
            />
          </MemoryRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`Should ButtonAddToList render correctly, !isFavorite, !isPromo`, () => {
    const tree = renderer
      .create(
          <MemoryRouter>
            <ButtonAddToList
              id={TEST_MOCKS.id}
              isFavorite={TEST_MOCKS.boolFalse}
              isPromo={TEST_MOCKS.boolFalse}
              onFavoriteClick={TEST_MOCKS.noop}
            />
          </MemoryRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

