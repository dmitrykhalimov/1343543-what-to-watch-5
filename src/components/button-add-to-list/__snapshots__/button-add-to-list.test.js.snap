// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonAddToList /> render Should ButtonAddToList render correctly, !isFavorite, !isPromo 1`] = `
<button
  className="btn btn--list movie-card__button"
  onClick={[Function]}
  type="button"
>
  <svg
    height="20"
    viewBox="0 0 19 20"
    width="19"
  >
    <use
      xlinkHref="#add"
    />
  </svg>
  <span>
    My list
  </span>
</button>
`;

exports[`<ButtonAddToList /> render Should ButtonAddToList render correctly, !isFavorite, isPromo 1`] = `
<button
  className="btn btn--list movie-card__button"
  onClick={[Function]}
  type="button"
>
  <svg
    height="20"
    viewBox="0 0 19 20"
    width="19"
  >
    <use
      xlinkHref="#add"
    />
  </svg>
  <span>
    My list
  </span>
</button>
`;

exports[`<ButtonAddToList /> render Should ButtonAddToList render correctly, isFavorite, !isPromo 1`] = `
<button
  className="btn btn--list movie-card__button"
  onClick={[Function]}
  type="button"
>
  <svg
    height="20"
    viewBox="0 0 19 20"
    width="19"
  >
    <use
      xlinkHref="#in-list"
    />
  </svg>
  <span>
    In list
  </span>
</button>
`;

exports[`<ButtonAddToList /> render Should ButtonAddToList render correctly, isFavorite, isPromo 1`] = `
<button
  className="btn btn--list movie-card__button"
  onClick={[Function]}
  type="button"
>
  <svg
    height="20"
    viewBox="0 0 19 20"
    width="19"
  >
    <use
      xlinkHref="#in-list"
    />
  </svg>
  <span>
    In list
  </span>
</button>
`;
